fe/fe-core/src/main/java/com/starrocks/sql/optimizer/rule/transformation/PruneHDFSScanColumnRule.java

```diff
@@ -138,6 +138,10 @@ private boolean containsMaterializedColumn(LogicalScanOperator scanOperator, Set
             return scanColumns.size() != 0 && !((LogicalHiveScanOperator) scanOperator).getPartitionColumns().containsAll(
                     scanColumns.stream().map(ColumnRefOperator::getName).collect(Collectors.toList()));
         }
+        if (scanOperator instanceof LogicalHudiScanOperator) {
```

Comment:
We definitely need our DLA new framework to not let this duplicated code exist.

Reply:
Yes, it is necessary and more abstract operaters should be extracted to decouple from concrete table properties.

Make a rough refinement about this here, PTAL

Reply:
> We definitely need our DLA new framework to not let this duplicated code exist.

that's true we should have a common prune rule which call api of connectors